scss/_patterns_accordion.scss

```diff
@@ -59,53 +61,93 @@
+        content: '';
+        display: inline-block;
+        height: $icon-size;
+        margin-right: $sph-inner;
+        width: $icon-size;
+      }
+    }
 
-    &::before {
-      background-size: $icon-size;
-      content: '';
-      display: inline-block;
-      height: $icon-size;
-      margin-right: $sph-inner;
-      width: $icon-size;
+    // FIXME: we shouldn't be targeting types directly here
+    // stylelint-disable selector-max-type
+    h2.p-accordion__title::before {
+      // make icon bigger for h2 (same as p-icon)
+      $icon-size-h2: $x-height;
+
+      background-size: $icon-size-h2;
+      height: $icon-size-h2;
+      width: $icon-size-h2;
     }
-  }
 
-  // FIXME: we shouldn't be targeting types directly here
-  // stylelint-disable selector-max-type
-  h2.p-accordion__title::before {
-    // make icon bigger for h2 (same as p-icon)
-    $icon-size-h2: $x-height;
+    h5.p-accordion__title::before,
+    h6.p-accordion__title::before {
+      // align icon within font metrics same as p-icon pattern
+      $vertical-offset: 0.5px;
 
-    background-size: $icon-size-h2;
-    height: $icon-size-h2;
-    width: $icon-size-h2;
-  }
+      vertical-align: calc(#{$vertical-offset} + #{0.5 * $cap-height} - #{0.5 * $icon-size});
+    }
+    // stylelint-enable selector-max-type
 
-  h5.p-accordion__title::before,
-  h6.p-accordion__title::before {
-    // align icon within font metrics same as p-icon pattern
-    $vertical-offset: 0.5px;
+    // override base expanded button styles
+    .p-accordion__tab--with-title[aria-expanded] {
+      background-color: inherit;
+    }
 
-    vertical-align: calc(#{$vertical-offset} + #{0.5 * $cap-height} - #{0.5 * $icon-size});
-  }
-  // stylelint-enable selector-max-type
+    .p-accordion__tab--with-title[aria-expanded='true'] .p-accordion__title::before {
+      @include vf-icon-minus($color-mid-dark);
+    }
 
-  // override base expanded button styles
-  .p-accordion__tab--with-title[aria-expanded] {
-    background-color: inherit;
+    .p-accordion__tab--with-title[aria-expanded='false'] .p-accordion__title::before {
+      @include vf-icon-plus($color-mid-dark);
+    }
   }
 
-  .p-accordion__tab--with-title[aria-expanded='true'] .p-accordion__title::before {
-    @include vf-icon-minus($color-mid-dark);
-  }
+  .p-accordion__heading {
+    margin-bottom: 0;
+    max-width: none;
+    padding-top: 0;
+
+    @at-root {
+      h2#{&} > .p-accordion__tab,
+      &.p-heading--two > .p-accordion__tab,
+      &.p-heading--2 > .p-accordion__tab {
+        $icon-top-position: calc(#{$table-cell-vertical-padding + (map-get($line-heights, h2) - $icon-size) * 0.5});
 
-  .p-accordion__tab--with-title[aria-expanded='false'] .p-accordion__title::before {
-    @include vf-icon-plus($color-mid-dark);
+        background-position-y: $icon-top-position;
+      }
+
+      h3#{&} > .p-accordion__tab,
+      &.p-heading--three > .p-accordion__tab,
+      &.p-heading--3 > .p-accordion__tab {
```

Comment:
Do we need to support `p-heading--` class names here or should using heading elements be enough?
I don't think we did support them in previous accordion implementation, we don't have examples for those (at least to test if they work as expected).

Allowing using them requires to mix component classes on one element (`<div class="p-heading--3 p-accordion__heading">`) which ideally we should not recommend.

Reply:
I would have thought we should - we allow those classes to be applied on standard heading elements, and I've found that ability useful on a number of occasions where, for example, a design requests a `h3` but in the page heirarchy it should semantically be a `h4`.

Reply:
Ok. I would just limit this to new `--3` class names, not the deprecated ones.

Reply:
:+1: I've updated with the rest of the feedback